org.lflang/src/org/lflang/generator/c/CCmakeGenerator.java

```diff
@@ -178,11 +195,25 @@ CodeBuilder generateCMakeCode(
             cMakeCode.pr("set(CMAKE_SYSTEM_NAME "+targetConfig.platformOptions.platform.getcMakeName()+")");
         }
 
-        cMakeCode.pr(setUpMainTarget.getCmakeCode(
-            hasMain,
-            executableName,
-            Stream.concat(additionalSources.stream(), sources.stream())
+        cMakeCode.pr("# Target definitions\n");
+        targetConfig.compileDefinitions.forEach((key, value) -> cMakeCode.pr(
+            "add_compile_definitions("+key+"="+value+")\n"
         ));
+        cMakeCode.newLine();
+
+        if (targetConfig.platformOptions.platform == Platform.ZEPHYR) {
```

Comment:
I'm OK with this special case for now, but I hope that it and the other `west`-specific code get factored out into the package manager eventually. It feels bad when branch statements accumulate and methods in and around the code generators get longer and more complex

Reply:
I very much agree with you on that